### PR TITLE
Add Protobuf communication through UART module

### PR DESCRIPTION
**Pull request by ecourtois**
Add `sys/uartpb` module, a module providing a generic class to send and receive Protocol messages over UART port.

This PR also adds an example that shows usage of a second UART to send and receive messages.
A client written in Python is also provided.

### CODE REVIEW

File: sys/uartpb/include/uartpb/UartProtobuf.hpp [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+        msg_length_bytes_received_ = 0;
+    }
+}
+
+template <typename InputMessageType, typename OutputMessageType,
+          void (*message_handler)(const InputMessageType &)>
+void UartProtobuf<InputMessageType, OutputMessageType, message_handler>::
+decode_message(uint32_t message_length, uint8_t *buffer)
+{
+    InputMessageType message;
```
Comment (gdoffe) [inline sys/uartpb/include/uartpb/UartProtobuf.hpp:195]: Noob question: as the template InputMessageType seems to be used only there. Could the template be defined for this function and not for the class ?

File: sys/uartpb/uartpb.cpp [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    uartpb->uart_rx_cb(data);
+}
+
+void *message_reader_wrapper(void *arg)
+{
+    UartProtobufBase *uartpb = (UartProtobufBase *)arg;
+    uartpb->message_reader();
+
+    return NULL;
+}
```
Comment (gdoffe) [inline sys/uartpb/uartpb.cpp:19]: I would appreciate a sequence diagram, i'm geting a bit lost. :confused:

File: .github/workflows/compilation.yml [hunk 3]
```diff
@@ -31,7 +31,7 @@ jobs:
     - name: Install toolchain
       run: |
         sudo apt-get update
-        sudo apt-get install -qq build-essential gcc-multilib g++-multilib gcc-arm-none-eabi libnewlib-arm-none-eabi uncrustify
+        sudo apt-get install -qq build-essential gcc-multilib g++-multilib gcc-arm-none-eabi libnewlib-arm-none-eabi uncrustify protobuf-compiler
```
Comment (gdoffe) [inline .github/workflows/compilation.yml:34]: should be in a dedicated commit

File: .gitignore [hunk 4]
```diff
@@ -62,3 +62,6 @@ scan-build/
 # Symbolic links
 pkg/*
 !pkg/vl53l0x-api
+
+# Protobuf generated files
+*_pb2.py
```
Comment (gdoffe) [inline .gitignore:67]: should be in a dedicated commit.

File: examples/uartpb/Makefile [hunk 5]
```diff
@@ -0,0 +1,15 @@
+APPLICATION = uart-nanopb
+
+# If no BOARD is found in the environment, use this default:
+BOARD ?= cogip-nucleo-f446re
+#BOARD ?= cogip-native
```
Comment (gdoffe) [inline examples/uartpb/Makefile:5]: dead code

File: examples/uartpb/README.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
+# Overview
```
Comment (gdoffe) [inline examples/uartpb/README.md:1]: See my comments in temporary PR: https://github.com/cogip/mcu-firmware/pull/49

**Review by gdoffe — CHANGES_REQUESTED** [review]

**Comment by ecourtois** [comment]
Close this PR as nanopb as finally been replaced by EmbeddedProto.
See #51

**Pull request closed without merging**